### PR TITLE
[ASCII-1430][E2E] Add helpers to wait for non-core agents to be ready

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Add a way to wait for non core agents (process, trace, security) to be ready.

### Motivation

As of now the e2e tests only wait for the core-agent to be ready as the framework doesn't know which agents will be started.
It will allow waiting for some other agents to be ready.

### Additional Notes

This was added on the `AgentClient` (which already contains the logic to wait for the core-agent).
Under the hood it tries to reach the status endpoint of each agent, and stops when they are all up.

Used an optional pattern, by default it doesn't wait for any agent, you can specify which agent to wait for and optionally on which port, as well as how long to wait for, and the path to the auth token (each option has sensible defaults so they are not required):
```
		awshost.WithAgentClientOptions(
			agentclientparams.WithAuthTokenPath(authTokenFilePath),
			agentclientparams.WithTraceAgent(),
			agentclientparams.WithProcessAgentOnPort(processCmdPort),
		),
```

### CODE REVIEW

File: test/new-e2e/pkg/components/remotehost_waitforagent.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+	return func(p *agentReadyParams) {
+		p.tick = d
+	}
+}
+
+func (p *agentReadyParams) processAgentStatusEndpoint() (string, bool) {
+	if p.processAgentPort == 0 {
+		return "", false
+	}
+	return fmt.Sprintf("http://localhost:%d/agent/status", p.processAgentPort), true
```
Comment (Kaderinho) [inline test/new-e2e/pkg/components/remotehost_waitforagent.go:170]: **❓ question:** what's the difference between `agent/status` and `/config/all` you mentioned in another PR ?‏

**Review by Kaderinho — APPROVED** [review]
LGTM, thank you for this 🎉 What would be needed to also support Windows ? I guess we need another curl function + change the default authPathToken but is there anything else ?

File: test/new-e2e/pkg/components/remotehost_waitforagent.go (hunk 1 above)
Comment (pgimalac) [reply test/new-e2e/pkg/components/remotehost_waitforagent.go:170]: Basically in the other PR I needed to fetch the config from every process so I used that endpoint directly, but this PR is more general so I used the status endpoints of each agent

**Comment by pgimalac** [comment]
Yeah for Windows there are three things that need an alternate implementation:
- the `curl` command
- the default `authTokenPath`
- the `cat` command to read the auth token file

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/pkg/components/remotehost_waitforagent.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package components
```
[the remark this answers is not in the document]
Comment (pgimalac) [reply test/new-e2e/pkg/components/remotehost_waitforagent.go:6]: The thing is that you can't add methods on a struct defined in another package
But I guess I could change the signature of `WaitForAgents` to take the `RemoteHost` as parameter, and add a method calling that one on the `RemoteHost` type
I'll give it a try !

File: test/new-e2e/pkg/utils/e2e/client/agent_client.go [hunk 3]
```diff
@@ -62,3 +70,88 @@ func NewDockerAgentClient(t *testing.T, docker *Docker, agentContainerName strin
...
+
+// waitForAgentsReady waits for the given non-core agents to be ready.
+// The given options configure which Agents to wait for, and how long to wait.
+//
+// Under the hood, this function checks the readiness of the agents by querying their status endpoints.
+// The function will wait until all agents are ready, or until the timeout is reached.
+// If the timeout is reached, an error is returned.
+//
+// As of now this is only implemented for Linux.
+func waitForAgentsReady(tt *testing.T, host *components.RemoteHost, params *agentclientparams.Params) {
```
Comment (pducolin) [inline test/new-e2e/pkg/utils/e2e/client/agent_client.go:82]: **❓ question**
Should this be a method of HostAgentClient instead, and have wait commands initialized at `NewHostAgentClient`?

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/pkg/utils/e2e/client/agent_client.go (hunk 3 above)
Comment (pgimalac) [reply test/new-e2e/pkg/utils/e2e/client/agent_client.go:82]: This is not running agent commands so we can't really work from `HostAgentClient` (it only runs agent subcommands) 😕

**Pull request merged**